dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md

```diff
@@ -0,0 +1,118 @@
+---
+id: 3000
+title: Maximum area of longest diagonal rectangle (Leetcode)
+sidebar_label: 3000-Maximum area of longest diagonal rectangle
+description: Solution of Maximum area of longest diagonal rectangle
+---
+
+## Problem Description
+
+| Problem Statement                                                                                                                   | Solution Link | LeetCode Profile |
+| :---------------------------------------------------------------------------------------------------------------------------------- | :------------ | :--------------- |
+| [Maximum area of longest diagonal rectangle](https://leetcode.com/problems/maximum-area-of-longest-diagonal-rectangle/description/) |
+
+## Problem Description
+
+You are given a 2D 0-indexed integer array dimensions.
+
+For all indices i, 0 <= i < dimensions.length, dimensions[i][0] represents the length and dimensions[i][1] represents the width of the rectangle i.
+
+Return the area of the rectangle having the longest diagonal. If there are multiple rectangles with the longest diagonal, return the area of the rectangle having the maximum area.
+
+### Examples
+
+#### Example 1
+
+- **Input:** $dimensions = [[9,3],[8,6]]$
+- **Output:** $48$
+- **Explanation:** $For index = 0, length = 9 and width = 3. Diagonal length = sqrt(9 * 9 + 3 * 3) = sqrt(90) ≈ 9.487.
+For index = 1, length = 8 and width = 6. Diagonal length = sqrt(8 * 8 + 6 * 6) = sqrt(100) = 10.
+So, the rectangle at index 1 has a greater diagonal length therefore we return area = 8 * 6 = 48..$
+
+#### Example 2
+
+- **Input:** $dimensions = [[3,4],[4,3]]$
+- **Output:** $12$
+- **Explanation:**$ Length of diagonal is the same for both which is 5, so maximum area = 12$.
+
+### Constraints
+
+- $1 <= dimensions.length <= 100$
```

Comment:
replace `$1 <= dimensions.length <= 100$` to `$1 \leq  dimensions.length \leq 100$`